SYCL/Matrix/joint_matrix_half.cpp

```diff
@@ -0,0 +1,165 @@
+//==-------- joint_matrix_half.cpp  - DPC++ joint_matrix------------ ----==//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+// REQUIRES: matrix
```

Comment:
The tests are effectively disabled due to this rule:
```
[2021-10-12T21:33:13.996Z] UNSUPPORTED: SYCL :: Matrix/joint_matrix_bf16.cpp (446 of 648)
[2021-10-12T21:33:13.996Z] UNSUPPORTED: SYCL :: Matrix/joint_matrix_half.cpp (447 of 648)
[2021-10-12T21:33:13.996Z] UNSUPPORTED: SYCL :: Matrix/joint_matrix_query_default.cpp (448 of 648)
```
Do we need to enable it by default in CI or it is not supported on most of the HW?

Reply:
Yes, the tests are running.

Reply:
Looking into the tests in the PR I see that all tests in the Matrix folder are skipped (marked UNSUPPORTED):
```
[2021-10-12T21:33:13.996Z] UNSUPPORTED: SYCL :: Matrix/joint_matrix_bf16.cpp (446 of 648)
[2021-10-12T21:33:13.996Z] UNSUPPORTED: SYCL :: Matrix/joint_matrix_half.cpp (447 of 648)
[2021-10-12T21:33:13.996Z] UNSUPPORTED: SYCL :: Matrix/joint_matrix_query_default.cpp (448 of 648)
[2021-10-12T21:33:13.996Z] UNSUPPORTED: SYCL :: Matrix/joint_matrix_ss_int8.cpp (449 of 648)
[2021-10-12T21:33:13.996Z] UNSUPPORTED: SYCL :: Matrix/joint_matrix_su_int8.cpp (450 of 648)
[2021-10-12T21:33:13.996Z] UNSUPPORTED: SYCL :: Matrix/joint_matrix_us_int8.cpp (451 of 648)
[2021-10-12T21:33:13.996Z] UNSUPPORTED: SYCL :: Matrix/joint_matrix_uu_int8.cpp (452 of 648)
```